### PR TITLE
Prlc100 227 featureflagging

### PR DESCRIPTION
**Pull request by yogendraU**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
PRLC100-227 C100 Rebuild feature flagging

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by channa-patil — APPROVED** [review]

File: charts/prl-citizen-frontend/values.preview.template.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ nodejs:
     CDAM_API_URL: 'https://ccd-case-document-am-api-prl-ccd-definitions-pr-676.service.core-compute-preview.internal'
     COS_URL: 'https://prl-cos-pr-562.service.core-compute-preview.internal'
     REDIS_HOST: 'prl-citizen-frontend-redis-aat.redis.cache.windows.net'
+    LAUNCH_DARKLY_OFFLINE_MODE: false
+    C100_REBUILD_FLAG: false
```
Comment (channa-patil) [inline charts/prl-citizen-frontend/values.preview.template.yaml:26]: @yogendraU Can we enable it in preview env by default ?

File: src/main/common/clients/launchDarklyClient.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      waitForInitialization: async (): Promise<any> => {
+        undefined;
+      },
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      variation: async (): Promise<any> => Promise.resolve({ testFlag: true }),
+    };
+  });
+
+  // test('Should initiate ldUser and client', function () {
```
Comment (channa-patil) [inline src/main/common/clients/launchDarklyClient.test.ts:33]: Can we please remove these commented lines

File: src/main/modules/oidc/index.ts [hunk 3]
```diff
@@ -59,6 +60,17 @@ export class OidcMiddleware {
...
+              req.session.c100RebuildLdFlag !== undefined
+                ? req.session.c100RebuildLdFlag
+                : (req.session.c100RebuildLdFlag = await getFeatureToggle().isC100reBuildEnabled());
+            if (c100RebuildLdFlag) {
+              return next();
+            } else {
+              return res.redirect(CITIZEN_HOME_URL);
+            }
+          }
           req.locals.C100Api = caseApi(req.session.user, req.locals.logger);
```
Comment (channa-patil) [inline src/main/modules/oidc/index.ts:74]: We can remove this duplicate invocation ? caseApi is being invoked above block on line#62 as well

File: config/development.yaml [hunk 4]
```diff
@@ -38,4 +38,11 @@ session:
 e2e:
   userTestPassword: ''
 appInsights:
-  instrumentationKey: false
\ No newline at end of file
+  instrumentationKey: false
+featureToggles:
+  c100Rebuild: C100_REBUILD_FLAG
```
Comment (channa-patil) [inline config/development.yaml:43]: Same here as well, we an enable it by default in Dev env I think

**Review by channa-patil — CHANGES_REQUESTED** [review]

File: charts/prl-citizen-frontend/values.preview.template.yaml (hunk 1 above)
Comment (yogendraU) [reply charts/prl-citizen-frontend/values.preview.template.yaml:26]: C100_REBUILD_FLAG has no significance in preview as flag is being calculated based on if LAUNCH_DARKLY_OFFLINE_MODE = false, then it fetches the flag from Launch Darkly API

File: src/main/common/clients/launchDarklyClient.test.ts (hunk 2 above)
Comment (yogendraU) [reply src/main/common/clients/launchDarklyClient.test.ts]: yes. they are removed in latest build now.

File: src/main/modules/oidc/index.ts (hunk 3 above)
Comment (yogendraU) [reply src/main/modules/oidc/index.ts:74]: Fixed it.

File: config/development.yaml (hunk 4 above)
Comment (yogendraU) [reply config/development.yaml]: Done.

**Pull request merged**